docs/ocis/development/debugging.md

```diff
@@ -179,8 +179,168 @@ For example `deployments/examples/ocis_wopi/docker-compose.yml`
+      - "traefik.http.routers.traefik.entrypoints=https"
+      - "traefik.http.routers.traefik.rule=Host(`${TRAEFIK_DOMAIN:-traefik.owncloud.test}`)"
+      - "traefik.http.routers.traefik.middlewares=traefik-auth"
+      - "traefik.http.routers.traefik.tls.certresolver=http"
+      - "traefik.http.routers.traefik.service=api@internal"
+    logging:
+      driver: ${LOG_DRIVER:-local}
+    restart: always
+
+  collabora:
+    image: collabora/code:23.05.5.2.1
+    networks:
+      ocis-net:
+    extra_hosts:
+      - "collaboration.owncloud.test:host-gateway"
+    environment:
+      aliasgroup1: https://collaboration.owncloud.test:9300
+      DONT_GEN_SSL_CERT: "YES"
+      extra_params: --o:ssl.enable=false --o:ssl.termination=true --o:welcome.enable=false --o:net.frame_ancestors=${OCIS_DOMAIN:-ocis.owncloud.test:9200}
+      username: ${COLLABORA_ADMIN_USER}
+      password: ${COLLABORA_ADMIN_PASSWORD}
+    cap_add:
+      - MKNOD
+    labels:
+      - "traefik.enable=true"
+      - "traefik.http.routers.collabora.entrypoints=https"
+      - "traefik.http.routers.collabora.rule=Host(`${COLLABORA_DOMAIN:-collabora.owncloud.test}`)"
+      - "traefik.http.routers.collabora.tls.certresolver=http"
+      - "traefik.http.routers.collabora.service=collabora"
+      - "traefik.http.services.collabora.loadbalancer.server.port=9980"
+      # websockets can't be opened when this is ommitted
+      - "traefik.http.middlewares.collabora.headers.customrequestheaders.X-Forwarded-Proto=https"
+      - "traefik.http.routers.collabora.middlewares=collabora"
+    logging:
+      driver: ${LOG_DRIVER:-local}
+    restart: always
+    healthcheck:
+      test: ["CMD", "curl", "-f", "http://localhost:9980/hosting/discovery"]
+
+volumes:
+  certs:
+  companion-data:
+
+networks:
+  ocis-net:
+
+```
+3.3 Run the docker compose `docker compose up -d`
+
+4. Run ocis or configure you IDE for debugging an application on the host machine
+``` bash
+OCIS_URL=https://ocis.owncloud.test:9200 \
+PROXY_ENABLE_BASIC_AUTH=true \
+MICRO_REGISTRY=nats-js-kv \
+MICRO_REGISTRY_ADDRESS="127.0.0.1:9233" \
+NATS_NATS_HOST="0.0.0.0" \
+NATS_NATS_PORT=9233 \
+GATEWAY_GRPC_ADDR="0.0.0.0:9142" \
+PROXY_CSP_CONFIG_FILE_LOCATION=ocis/services/proxy/pkg/config/csp.yaml \
+PROXY_ENABLE_BASIC_AUTH=true \
+OCIS_EXCLUDE_RUN_SERVICES=app-provider \
+ocis/bin/ocis server
+```
+
+5. Run collaboration server or configure you IDE for debugging an application on the host machine
+```bash
+COLLABORATION_APP_NAME=Collabora \
+COLLABORATION_APP_ICON=https://collabora.owncloud.test/favicon.ico \
+COLLABORATION_CS3API_DATAGATEWAY_INSECURE=true \
+COLLABORATION_HTTP_ADDR=collaboration.owncloud.test:9300 \
+COLLABORATION_HTTP_SCHEME=http \
+COLLABORATION_LOG_LEVEL=debug \
+COLLABORATION_WOPIAPP_ADDR=https://collabora.owncloud.test \
+COLLABORATION_WOPIAPP_INSECURE=true \
```

Comment:
some of these env vars will change with 
- https://github.com/owncloud/ocis/pull/9165
- https://github.com/owncloud/ocis/pull/9277
- https://github.com/owncloud/ocis/pull/9280

Reply:
Reworked the debugging doc. Added the collaboration docker compose files